File: snippets/rust/batch_update.rs
use qdrant_client::qdrant::{
    points_selector::PointsSelectorOneOf,
    points_update_operation::{
        Operation, PointStructList, SetPayload, UpdateVectors,
    },
    PointStruct, PointVectors, PointsIdsList, PointsSelector, PointsUpdateOperation,
};

let client = QdrantClient::from_url("http://localhost:6334").build()?;

client
    .batch_updates_blocking(
        "{collection_name}",
        &[
            PointsUpdateOperation {
                operation: Some(Operation::Upsert(PointStructList {
                    points: vec![PointStruct::new(
                        1,
                        vec![1.0, 2.0, 3.0, 4.0],
                        json!({}).try_into().unwrap(),
                    )],
                    ..Default::default()
                })),
            },
            PointsUpdateOperation {
                operation: Some(Operation::UpdateVectors(UpdateVectors {
                    points: vec![PointVectors {
                        id: Some(1.into()),
                        vectors: Some(vec![1.0, 2.0, 3.0, 4.0].into()),
                    }],
                    ..Default::default()
                })),
            },
            PointsUpdateOperation {
                operation: Some(Operation::OverwritePayload(SetPayload {
                    points_selector: Some(PointsSelector {
                        points_selector_one_of: Some(PointsSelectorOneOf::Points(
                            PointsIdsList {
                                ids: vec![1.into()],
                            },
                        )),
                    }),
                    payload: HashMap::from([("test_payload".to_string(), 1.into())]),
                    ..Default::default()
                })),
            },
        ],
        None,
    )
    .await?;

File: snippets/go/batch_update.go
package client

import (
	"context"

	"github.com/qdrant/go-client/qdrant"
)

func batchUpdate() {
	client, err := qdrant.NewClient(&qdrant.Config{
		Host: "localhost",
		Port: 6334,
	})
	if err != nil {
		panic(err)
	}

	_, err = client.UpdateBatch(context.Background(), &qdrant.UpdateBatchPoints{
		CollectionName: "{collection_name}",
		Operations: []*qdrant.PointsUpdateOperation{
			qdrant.NewPointsUpdateUpsert(&qdrant.PointsUpdateOperation_PointStructList{
				Points: []*qdrant.PointStruct{{
					Id:      qdrant.NewIDNum(1),
					Vectors: qdrant.NewVectors(1.0, 2.0, 3.0, 4.0),
				}},
			}),
			qdrant.NewPointsUpdateUpdateVectors(&qdrant.PointsUpdateOperation_UpdateVectors{
				Points: []*qdrant.PointVectors{
					{
						Id:      qdrant.NewIDNum(1),
						Vectors: qdrant.NewVectors(0.1, 0.2, 0.3, 0.4),
					},
				},
			}),
			qdrant.NewPointsUpdateSetPayload(&qdrant.PointsUpdateOperation_SetPayload{
				PointsSelector: qdrant.NewPointsSelector(qdrant.NewIDNum(1)),
				Payload: qdrant.NewValueMap(map[string]any{
					"test_payload_2": 2,
					"test_payload_3": 3,
				}),
			}),
		},
	})
	if err != nil {
		panic(err)
	}
}
